src/js/references/hooks.ts

```diff
@@ -128,3 +130,55 @@ export function useUpdateSourceTypes(
         register,
     };
 }
+
+/**
+ * Check if the logged in account has the passed `right` on the reference detail is loaded for.
+ *
+ * @param reference - The reference to check
+ * @param right - The right to check for (eg. modify_otu)
+ * @returns Whether the right is possessed by the account
+ */
+export function useCheckReferenceRight(reference: Reference, right: string) {
```

Comment:
```suggestion
export function useCheckReferenceRight(referenceId: string, right: ReferenceRight) {
```
Define an enum for this, prevents typo errors. I would also just pass the ref string. 

I am also personally leaning towards just passing the id in these requests, keeps the data flowing one way and imo makes it easier to reason through what happens at given load steps.